src/components/__tests__/search-bar-test.jsx

```diff
@@ -16,20 +16,42 @@ describe('search-bar', () => {
   });
 
   it('Handles search queries with a protocol correctly', () => {
+    const onSearch = jest.fn();
     const searchBar = shallow(<SearchBar onSearch={onSearch} />);
     const searchBarInput = searchBar.find('input');
     searchBarInput.simulate('change', {target: {value: 'http://epa'}});
 
     jest.runAllTimers();
-    expect(onSearch).toHaveBeenCalledWith({url: 'http://epa*'});
+    expect(onSearch).toHaveBeenCalledWith({endDate: null, startDate: null, url: 'http://epa*'});
   });
 
   it('Handles search queries without a protocol correctly', () => {
+    const onSearch = jest.fn();
     const searchBar = shallow(<SearchBar onSearch={onSearch} />);
     const searchBarInput = searchBar.find('input');
     searchBarInput.simulate('change', {target: {value: 'epa'}});
 
     jest.runAllTimers();
-    expect(onSearch).lastCalledWith({url: '*//epa*'});
+    expect(onSearch).toHaveBeenCalledWith({endDate: null, startDate: null, url: '*//epa*'});
+  });
+
+  it('Handles date range search queries for startDate', () => {
+    const onSearch = jest.fn();
+    const searchBar = mount(<SearchBar onSearch={onSearch} />);
+    searchBar.find('.DateInput_input.DateInput_input_1').first().simulate('focus');
```

Comment:
I know these tests are brittle/annoying because they are relying on the classes from the library, so if you had any other suggestions as to how to handle this, lmk.

Reply:
Yeah, I don’t think there are really better options here, so this is OK. Some techniques I use in this kind of situation:

- Lightweight: add a comment noting that this part is expected to be brittle, so it’s not totally confusing when it breaks for someone when updating the dependency. :)

- Midweight: wrap the logic of getting the input (or finding a day button, or whatever) so it’s more readable:

    ```js
    function getDateField (searchBar, inputType = 'FROM') {
      // NOTE: this is prone to break with updates to react-dates.
      const index = inputType === 'FROM' ? 0 : 1;
      return searchBar.find(`.DateInput_input.DateInput_input_1`).at(index);
    }

    it('does something I want to test', () => {
      const searchBar = mount(<SearchBar onSearch={onSearch} />);
      getDateField(searchBar, 'FROM').simulate('focus');
      getDateField(searchBar, 'TO').simulate('focus');
      expect(something);
    }
    ```

- Heavyweight: add a method to the `SearchBar` class to get the components of the calendar (you’d have to use refs for this). Almost certainly overkill in most cases.

Reply:
OH! I totally missed that `<DateRangePicker>` requires you to specify IDs for these! Maybe you should just make those a class or instance property you can get at from the test? e.g:

```js
// Where you define the class, make the IDs something you can get:
class SearchDatePicker extends React.Component {
  static startInputId = 'startDate'
  static endInputId = 'endDate'

  render () {
    return (
      <DateRangePicker
        startDateId={SearchDatePicker + this.instanceId}
        endDateId={SearchDatePicker + this.instanceId}
        ... other props ...
      />
    );
  }
}

// In the test, use that!
function getDateField (searchBar, inputType = 'start') {
  // This is all public API and not brittle!
  const id = SearchDatePicker[`${inputType}DateId`];
  return searchBar.find(`#{id}`).first();
}
```